connect/runtime/src/main/java/org/apache/kafka/connect/storage/KafkaConfigBackingStore.java

```diff
@@ -761,6 +803,44 @@ public void onCompletion(Throwable error, ConsumerRecord<String, byte[]> record)
 
     }
 
+    @SuppressWarnings("unchecked")
+    RestartRequest recordToRestartRequest(ConsumerRecord<String, byte[]> record, SchemaAndValue value) {
+        String connectorName = record.key().substring(RESTART_PREFIX.length());
+        if (value.value() == null) {
+            log.error("Ignoring restart request because it is unexpectedly null");
+            return null;
+        }
+        if (!(value.value() instanceof Map)) {
+            log.error("Ignoring restart request because the value is not a Map but is {}", value.value().getClass());
+            return null;
+        }
```

Comment:
I see we've been verbose in similar logic above before, but maybe we can improve on that a bit, at least in new code. Here's another suggestion: 
```suggestion
        if (!(value.value() instanceof Map)) {
            log.error("Ignoring restart request because the value is not a Map but is {}", value.value() == null ? "null" : value.value().getClass());
            return null;
        }
```

Reply:
This misuse might have originated because above, specifically for tasks, `null` is actually a valid value (tombstone/delete), which is not the case here.